code/game/objects/items/devices/scanners_vr.dm

```diff
@@ -0,0 +1,209 @@
+	output += "<b>Body-Scan (One Time): </b>\[<a href='?src=\ref[src];target=\ref[H];bodyscan=1'>Perform</a>\]<br>"
+
+	//Saving a mind
+	output += "<b>Store Full Mind: </b>"
+	if(stored_mind)
+		output += "<span class='notice'>Already Stored</span> ([stored_mind.name])<br>"
+	else if(H.mind)
+		output += "\[<a href='?src=\ref[src];target=\ref[H];mindsteal=1'>Perform</a>\]<br>"
+	else
+		output += "<span class='warning'>Unable</span><br>"
+
+	//Soulcatcher transfer
+	if(stored_mind && H.nif)
+		var/datum/nifsoft/soulcatcher/SC = H.nif.imp_check(NIF_SOULCATCHER)
+		if(SC)
+			output += "<b>Store in Soulcatcher: </b>\[<a href='?src=\ref[src];target=\ref[H];mindput=1'>Perform</a>\]<br>"
+
+	to_chat(user,output)
+
+/obj/item/device/sleevemate/Topic(href, href_list)
+	usr.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
+
+	//Sanity checking/href-hacking checking
+	if(usr.get_active_hand() != src)
+		to_chat(usr,"<span class='warning'>You're not holding \the [src].</span>")
+		return
+
+	var/target_ref = href_list["target"]
+	var/mob/living/target = locate(target_ref) in mob_list
+	if(!target)
+		to_chat(usr,"<span class='warning'>Unable to operate on that target.</span>")
+		return
+
+	if(!usr.Adjacent(target))
+		to_chat(usr,"<span class='warning'>You are too far from that target.</span>")
+		return
+
+	//The actual options
+	if(href_list["mindscan"])
+		if(!target.mind)
+			to_chat(usr,"<span class='warning'>Target seems totally braindead.</span>")
+			return
+
+		var/nif
+		if(ishuman(target))
+			var/mob/living/carbon/human/H = target
+			nif = H.nif
+
+		usr.visible_message("[usr] begins scanning [target]'s mind.","<span class='notice'>You begin scanning [target]'s mind.</span>")
+		if(do_after(usr,8 SECONDS,target))
+			transcore.m_backup(target.mind,nif,one_time = TRUE)
+			to_chat(usr,"<span class='notice'>Mind backed up!</span>")
+		else
+			to_chat(usr,"<span class='warning'>You must remain close to your target!</span>")
+
+		return
+
+	if(href_list["bodyscan"])
+		if(!ishuman(target))
+			to_chat(usr,"<span class='warning'>Target is not of an acceeptable body type.</span>")
+			return
+
+		var/mob/living/carbon/human/H = target
+
+		usr.visible_message("[usr] begins scanning [target]'s body.","<span class='notice'>You begin scanning [target]'s body.</span>")
+		if(do_after(usr,8 SECONDS,target))
+			var/datum/transhuman/body_record/BR = new()
+			BR.init_from_mob(H, TRUE, TRUE)
+			to_chat(usr,"<span class='notice'>Body scanned!</span>")
+		else
+			to_chat(usr,"<span class='warning'>You must remain close to your target!</span>")
+
+		return
+
+	if(href_list["mindsteal"])
+		if(!target.mind)
+			to_chat(usr,"<span class='warning'>Target seems totally braindead.</span>")
+			return
+
+		usr.visible_message("<span class='warning'>[usr] begins downloading [target]'s mind!</span>","<span class='notice'>You begin downloading [target]'s mind!</span>")
+		if(do_after(usr,35 SECONDS,target)) //This is powerful, yo.
+			if(!stored_mind)
```

Comment:
True true! But still, they might have ghostized in some other way! Someone else with a sleevemate or them ghostfacing or something right?